include/mixer.h

```diff
@@ -119,6 +122,13 @@ class MixerChannel {
 	void Mix(int _needed);
 	void AddSilence(); // Fill up until needed
 
+	void EnableLowPassFilter(const bool enabled = true);
+	void ForceLowPassFilter(const bool force = true);
+	void ConfigureLowPassFilter(const uint8_t order, const uint16_t cutoff_freq);
+
+	void EnableZeroOrderHoldUpsampler(const bool enabled = true);
```

Comment:
There's a log of existing DOSBox code that takes in bool arguments (and defaults on top of that). 

When literals are used, the functions get hard to understand, and when defaults are thrown in they're even more confusing.

Take this example:

``` c++
file.writeData( data, true);
file.writeData( data);
file.writeData( data, false);
```

Now compare that to:

``` c++
enum class WriteMode {
  Append,
  Overwrite
};

file.writeData( data, WriteMode::Append);
file.writeData( data, WriteMode::Overwrite);
```

In this case, I'd suggest a general filter state enum that has on, off, or forced-on states.
It could be put in `mixer.h` and used quite broadly, I think.

Reply:
Yeah, I was following existing pattern, but I think it's a good idea. The situation is even worse in C++ compared to other languages because of the lack or named arguments, so this will make the code clearer.